lib/ansible/modules/cloud/amazon/iam_user_info.py

```diff
@@ -0,0 +1,184 @@
+    description: list of maching iam users
+    returned: success
+    type: complex
+    contains:
+        arn:
+            description: the ARN of the user
+            returned: if user exists
+            type: str
+            sample: "arn:aws:iam::156360693172:user/dev/test_user"
+        create_date:
+            description: the datetime user was created
+            returned: if user exists
+            type: str
+            sample: "2016-05-24T12:24:59+00:00"
+        password_last_used:
+            description: the last datetime the password was used by user
+            returned: if password was used at least once
+            type: str
+            sample: "2016-05-25T13:39:11+00:00"
+        path:
+            description: the path to user
+            returned: if user exists
+            type: str
+            sample: "/dev/"
+        user_id:
+            description: the unique user id
+            returned: if user exists
+            type: str
+            sample: "AIDUIOOCQKTUGI6QJLGH2"
+        user_name:
+            description: the user name
+            returned: if user exists
+            type: str
+            sample: "test_user"
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, camel_dict_to_snake_dict, ec2_argument_spec, get_aws_connection_info
+
+try:
+    import botocore
+    from botocore.exceptions import ClientError, ParamValidationError
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+
+def list_iam_users(connection, module):
+
+    name = module.params.get('name')
+    group = module.params.get('group')
+    path = module.params.get('path')
+
+    params = dict(MaxItems=1000)
+    iam_users = []
+
+    if not group and not path:
+        if name:
+            params['UserName'] = name
+        try:
+            iam_users.append(connection.get_user(**params)['User'])
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json_aws(e, msg="Couldn't get IAM user info for user %s" % name)
+
+    if group:
+        params['GroupName'] = group
+        try:
+            iam_users = connection.get_group(**params)['Users']
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json_aws(e, msg="Couldn't get IAM user info for group %s" % group)
+        if name:
+            iam_users = [user for user in iam_users if user['UserName'] == name]
+
+    if path and not group:
+        params['PathPrefix'] = path
+        try:
+            iam_users = connection.list_users(**params)['Users']
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json_aws(e, msg="Couldn't get IAM user info for path %s" % path)
+        if name:
+            iam_users = [user for user in iam_users if user['UserName'] == name]
+
+    module.exit_json(iam_users=[camel_dict_to_snake_dict(user) for user in iam_users])
+
+
+def main():
+    argument_spec = ec2_argument_spec()
```

Comment:
AnsibleAWSModule automatically adds the EC2 options, you don't need to use ec2_argument_spec any more

Reply:
https://github.com/ansible/ansible/pull/23382/commits/450fd12d0898960b65177a0d10f14765fa39d5fc removes this.